src/webhook_launcher/app/tasks.py

```diff
@@ -87,20 +93,28 @@ def trigger_build(mapobj, user, lsr=None, tag=None, force=False):
                 delayed = True
                 break
 
+    # Updat LastSeenRevision.tag if this is going to be built
+    if build and not skipped and tag:
```

Comment:
Why not fold this within the existing `if build:`? Or then with my comment below about calling handle_build even break out earlier.